.gitlab-ci.yml

```diff
@@ -1,37 +1,591 @@
+  only:
+    variables:
+      - $BUILD_RASPBERRYPI3 == "true"
+      - $TEST_RASPBERRYPI3 == "true"
+
+.template_test_integ: &test_integration
+  image: docker:dind
+  tags:
+    - mender-qa-slave
+  before_script:
+    - /usr/local/bin/dockerd-entrypoint.sh &
+    - sleep 1
+    - docker version
+    - apk --update --no-cache add bash curl git openssh device-mapper py-pip
+      iptables gcc python2-dev libffi-dev libc-dev openssl-dev make python3
+      python3-dev e2fsprogs-extra openssl
+    # mender-artifact and other GO tools not build for muslc will throw "not found" error
+    # see https://stackoverflow.com/questions/34729748/installed-go-binary-not-found-in-path-on-alpine-linux-docker
+    - mkdir /lib64 && ln -s /lib/libc.musl-x86_64.so.1 /lib64/ld-linux-x86-64.so.2
+    - pip install docker-compose==1.24.0
+    - pip3 install pyyaml
+    - rm -rf /var/cache/apk/*
+    - git clone https://github.com/mendersoftware/integration || true
+    - (cd integration && git fetch -u -f origin ${INTEGRATION_REV}:pr && git checkout pr)
+    - pip install pycrypto && pip install -r integration/tests/requirements.txt
+    - pip install pytest-xdist --upgrade
+    - pip install pytest-html --upgrade
+    # Load all docker images
+    - docker load -i deployments.tar
+    - docker load -i deviceadm.tar
+    - docker load -i deviceauth.tar
+    - docker load -i gui.tar
+    - docker load -i inventory.tar
+    - docker load -i api-gateway.tar
+    - docker load -i mender-conductor.tar
+    - docker load -i mender-conductor-enterprise.tar
+    - docker load -i tenantadm.tar
+    - docker load -i useradm.tar
+    - docker load -i email-sender.tar
+    - docker load -i mender-client-docker.tar
+    - docker load -i mender-client-qemu_qemux86_64_uefi_grub.tar ||
+      docker load -i mender-client-qemu_vexpress_qemu.tar
+    # Login for private repos
+    - docker login -u menderbuildsystem -p ${DOCKER_PASSWORD}
+    # Set testing versions to PR
+    - integration/extra/release_tool.py --set-version-of deployments --version pr
+    - integration/extra/release_tool.py --set-version-of deviceadm --version pr
+    - integration/extra/release_tool.py --set-version-of deviceauth --version pr
+    - integration/extra/release_tool.py --set-version-of gui --version pr
+    - integration/extra/release_tool.py --set-version-of inventory --version pr
+    - integration/extra/release_tool.py --set-version-of mender --version pr
+    - integration/extra/release_tool.py --set-version-of mender-api-gateway-docker --version pr
+    - integration/extra/release_tool.py --set-version-of mender-conductor --version pr
+    - integration/extra/release_tool.py --set-version-of mender-conductor-enterprise --version pr
+    - integration/extra/release_tool.py --set-version-of tenantadm --version pr
+    - integration/extra/release_tool.py --set-version-of useradm --version pr
+    - integration/extra/release_tool.py --set-version-of mender-client-qemu --version pr
```

Comment:
Should use a loop here, like:

```
for repo in `integration/extra/release_tool.py -l`; do
    integration/extra/release_tool.py --set-version-of $repo --version pr
done
```

This way we won't ever miss a version that we add later. And it will reveal if we have missed it anywhere else (there will be `pr` tag which can't be pulled).

Reply:
Thanks! I'll rework

Side note: we could also have something like `--set-version-of -a --version pr`

Reply:
@kacf Very good catch! I used `-l docker -a` options, so that we get the docker images and not the repos.

We were missing some of them (i.e. api-gateway, email-sender, mender-client-docker). And I think some of the failing integration tests are related to this.